pkg/runtime/reconciler.go

```diff
@@ -1100,7 +1102,7 @@ func (r *resourceReconciler) getOwnerAccountID(
 		return ackv1alpha1.AWSAccountID(accID), true
 	}
 
-	controllerAccountID := ackv1alpha1.AWSAccountID(r.cfg.AccountID)
+	controllerAccountID := ackv1alpha1.AWSAccountID(r.cfg.DefailtAccountID)
 	// look for owner account id in the resource status
 	acctID := res.Identifiers().OwnerAccountID()
 	if acctID != nil {
```

Comment:
This block is a little worrying:
```go
acctID := res.Identifiers().OwnerAccountID()
	if acctID != nil {
		return *acctID, *acctID != controllerAccountID
	}
```
I'If `controllerAccountID` is empty, we will return true, which doesn't match the logic in the main reconcile function.

Reply:
I'm not sure what you mean here

Reply:
You have a change that depends on the boolean (*acctID != controllerAccountID):
```go
	} else if !needCARMLookup && acctID == "" {
		return ctrlrt.Result{}, fmt.Errorf("unable to locate AWS credentials to manage resource: both controller credentials and a proper CARM configuration are missing")
	}
```

if `*controllerAccountID = ""` then `*acctID != controllerAccountID` will be true which impacts the entry `!needCARMLookup && acctID == ""`

Reply:
what could the value of `accID` be in that case?